testcases/crypto/ec_func.c

```diff
@@ -162,6 +162,566 @@ _signVerifyParam signVerifyInput[] = {
+
+        rc = funcs->C_GenerateKeyPair(session, &mech, pub_attr, pub_attr_len, prv_attr, prv_attr_len, &publ_keyA, &priv_keyA);
+        if (rc != CKR_OK) {
+            testcase_fail("C_GenerateKeyPair with valid input failed at i=%lu, rc=%s", i, p11_get_ckr(rc));
+            goto testcase_cleanup;
+        }
+
+        // Extract public key A
+        rc = funcs->C_GetAttributeValue(session, publ_keyA, extr1_tmpl, extr1_tmpl_len);
+        if (rc != CKR_OK) {
+            testcase_error("C_GetAttributeValue #1: rc = %s", p11_get_ckr(rc));
+            goto testcase_cleanup;
+        }
+
+        // Now generate the EC key pair for party B
+        mech.mechanism = CKM_EC_KEY_PAIR_GEN;
+        mech.ulParameterLen = 0;
+        mech.pParameter = NULL;
+
+        rc = funcs->C_GenerateKeyPair(session, &mech, pub_attr, pub_attr_len, prv_attr, prv_attr_len, &publ_keyB, &priv_keyB);
+        if (rc != CKR_OK) {
+            testcase_fail("C_GenerateKeyPair with valid input failed at i=%lu, rc=%s", i, p11_get_ckr(rc));
+            goto testcase_cleanup;
+        }
+
+        // Extract public key B
+        rc = funcs->C_GetAttributeValue(session, publ_keyB, extr2_tmpl, extr2_tmpl_len);
+        if (rc != CKR_OK) {
+            testcase_error("C_GetAttributeValue #1: rc = %s", p11_get_ckr(rc));
+            goto testcase_cleanup;
+        }
+
+        // Check if the key lengths are equal
+        if (extr1_tmpl->ulValueLen != extr2_tmpl->ulValueLen) {
+            testcase_error("Length of public key A not equal to length of public key B");
+            goto testcase_cleanup;
+        }
+
+        // Testcase #2 - Now derive the secrets...
+
+        for (int j=0; j < NUM_KDFS; j++) {
+
+            for (int k=0; k<NUM_SECRET_KEY_LENGTHS; k++) {
+
+                if (too_many_key_bytes_requested(i, j, secret_key_len[k]) ||
+                    too_many_key_bytes_requested_ep11(i, j, secret_key_len[k])) {
+                    testcase_skip("Cannot provide %ld key bytes with curve %i without a kdf.\n", secret_key_len[k], i);
+                    continue;
+                }
+
+                CK_ATTRIBUTE  secretA_tmpl[] = {
+                    {CKA_VALUE, secretA_value, sizeof(secretA_value)},
+                };
+                CK_ULONG secretA_tmpl_len = sizeof(secretA_tmpl)/sizeof(CK_ATTRIBUTE);
+
+                CK_ATTRIBUTE  secretB_tmpl[] = {
+                    {CKA_VALUE, deriveB_value, sizeof(deriveB_value)},
+                };
+                CK_ULONG secretB_tmpl_len = sizeof(secretB_tmpl)/sizeof(CK_ATTRIBUTE);
+
+                CK_ATTRIBUTE  derive_tmpl[] = {
+                    {CKA_CLASS, &class, sizeof(class)},
+                    {CKA_KEY_TYPE, &key_type, sizeof(key_type)},
+                    {CKA_VALUE_LEN, &(secret_key_len[k]), sizeof(CK_ULONG)},
+                };
+                CK_ULONG secret_tmpl_len = sizeof(derive_tmpl)/sizeof(CK_ATTRIBUTE);
+
+                for (int m=0; m<NUM_SHARED_DATA; m++) {
```

Comment:
Do not declare int m inside the for, please do it outside. This is causing Travis CI build to fail.